libkbfs/modes.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2018 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+// NewInitModeFromType returns an InitMode object corresponding to the
```

Comment:
Did you intend to have this at beginning of the file?

Reply:
Yeah, I kind of consider it a constructor, which I usually put at the top for easy access.  But I could be convinced otherwise.